### PR TITLE
Add ServicePlanBlocks struct

### PR DESCRIPTION
**Pull request by jfurrow**
This will be used in a future PR

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @daiyiwhale

File: src/js/structs/ServicePlanBlocks.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      if (!(block instanceof ServicePlanBlock)) {
+        block = new ServicePlanBlock(block);
+      }
+
+      if (block.isInProgress()) {
+        this._activeBlocks.push(block);
+      }
+    });
+
+    this._activeBlockCount = this._activeBlocks.length;
```
Comment (MatApple) [inline src/js/structs/ServicePlanBlocks.js]: We probably don't need to store this right?

File: src/js/structs/ServicePlanBlocks.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  }
+
+  getActive() {
+    if (this._activeBlockCount === 0) {
+      return null;
```
Comment (MatApple) [inline src/js/structs/ServicePlanBlocks.js]: In this instance would it make more sense to just return an empty array?

**Comment by MatApple** [comment]
LGTM after those comments!

File: src/js/structs/ServicePlanBlocks.js (hunk 2 above)
Comment (jfurrow) [inline src/js/structs/ServicePlanBlocks.js]: Done!

File: src/js/structs/ServicePlanBlocks.js (hunk 1 above)
Comment (jfurrow) [inline src/js/structs/ServicePlanBlocks.js]: Done!

**Comment by jfurrow** [comment]
@MatApple I also added an aggregator for complete blocks which should work identically to active blocks.

**Comment by MatApple** [comment]
Looks good! Will wait for tests

**Pull request merged**